packages/template-retail-react-app/app/components/store-availability-text/index.jsx

```
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useContext} from 'react'
+import PropTypes from 'prop-types'
+import {FormattedMessage} from 'react-intl'
+
+// Components
+import {Box, Link} from '@salesforce/retail-react-app/app/components/shared/ui'
+
+// Others
+import {StoreLocatorContext} from '@salesforce/retail-react-app/app/components/store-locator-modal'
+
+const StoreAvailabilityText = ({productInventories}) => {
+    const {selectedStore, isStoreSelected} = useContext(StoreLocatorContext)
+    const inStock =
+        selectedStore.inventoryId &&
+        Boolean(
+            productInventories?.find(
+                (inventory) => inventory.id === selectedStore.inventoryId && inventory.orderable
+            )
+        )
+
+    return (
+        <Box gap={1} fontWeight={400} display="flex">
+            {isStoreSelected ? (
+                <>
+                    {inStock ? (
+                        <FormattedMessage
+                            id={'product_view.store_availability.in_stock_at'}
+                            defaultMessage={'In Stock at'}
+                        />
+                    ) : (
+                        <FormattedMessage
+                            id={'product_view.store_availability.out_of_stock_at'}
+                            defaultMessage={'Out of Stock at'}
+                        />
+                    )}
+
+                    <Link>{selectedStore.name}</Link>
```

it's possible to create a store with no name. Should we throw an error in that case?

---

If a store is created with no name, that should be an error thrown on the BM/store creation side and not something that the shopers' UX should have to deal with with an error. I think it would be fine to keep it simple and just display what the API gives.